PR: https://github.com/odoo/odoo/pull/

From: ea66192276921be1bafe73255fa77c55f3e10ac7
From: Haresh Shyara
Date: 2017-08-09 08:03:08

Breaking data model changes score: 6, change matches:
-    _inherit = 'sale.config.settings'
-    module_sale_subscription = fields.Boolean("Subscriptions")
-    module_sale_stock = fields.Boolean("Inventory Management")
-    module_sale_timesheet = fields.Boolean("Timesheets")
-    module_sale_ebay = fields.Boolean("eBay")
-    module_web_clearbit = fields.Boolean("Customer Autocomplete")

Total Changes: 12

[REF] crm,sale: Split the applications into 2 separate apps

Purpose
=======

You install CRM but you don't see CRM icon (except in English, as long as sales is not installed) -> confusing
The dashboard with salesteam is not obvious & not useful for simple users: what are those channel boxes about? why don't I see all my activities?, etc.

Specifications
==============

Split CRM & Sales + land in business views straight away

Default Entry Views
~~~~~~~~~~~~~~~~~~~

* New default view for CRM: Pipeline kanban
* New default view for Sales: Quotation list

Dashboards
~~~~~~~~~~

* Move sales teams dashboard to Reporting: Sales Channels.
* Remove the upper part of the sales team dashboard

Menuitems Structure
~~~~~~~~~~~~~~~~~~~

* CRM menu

Pipeline:
	- Leads (optional)
	- Pipeline (default menu, default filter "My Pipeline")
	- Next Activities -> to remove (now there is an icon in top bar for next activities)
	- Quotations (If sale_management installed)
Customers
Phone Calls (if voip, if not yet removed because there is a BE task to use next activities instead)
Leads Management:
	- Scoring Rules
	- Leads Assignation
	- Team Assignation
Reporting:
	- Leads
	- Scoring page views
	- Opp. Assignement
	- Patnerships
	- Pipeline
	- Activities
	- Phonecalls
	- Sales Channels
Configuration:
	- Settings
	- Sales Channels
	- Activity Types
	- Leads & Opportunities:
		- Lead Tags
		- Lost Reasons
	- Resellers
		- Partner Level
		- Partner Activations
* Sales menu:

Orders:
	- Quotations
	- Orders
	- Customers
Invoicing:
	- Orders to Invoice
	- Orders to Upsell
Catalog:
	- Products
	- Promotion Programs
	- Coupon Programs
Reporting:
	- Sales
	- Sales Channels
	- All channels sales orders
Configuration:
	- Settings
	- Sales Channels
	- Sales Orders:
		- Quotation templates
		- Payment acquirers
		- Delivery Methods

Settings
~~~~~~~~

* Split settings form
* Remove recommended apps section
* Remove Timesheets section
* Remove Inventory Management and move shipping connectors to Shipping section
* Remove Integrations section
	- Move Docsaway option to "Quotations & Orders", right after proforma
	- Autocomplete and asterix: move to CRM section
* Restructure and rename sections this way:
	- Product Catalog
	- Pricing
	- Quotations & Orders
	- Shipping
	- Invoicing
	- eBay (if installed)
* Remove all the "Save this page and come back here" when checking a box that installs a new module

* restructure a bit the CRM settings, with 3 sections:

Pipeline
	- Leads
Contacts
	- Phone Validation
	- Customer Autocomplete
Integrations
	- Google Calendar / Synchronize your calendar with Google Calendar (copy from general settings but don't show message "Save this page and come back here...".
	- Asterisk (VOIP)

================================= pseudo patch: =================================

--- a/addons/sale/models/sale_config_settings.py
+++ b/addons/sale/models/sale_config_settings.py
@@ -5,8 +5,11 @@ from odoo import api, fields, models
 
 
 class SaleConfigSettings(models.TransientModel):
-    _inherit = 'sale.config.settings'
+    _name = 'sale.config.settings'
+    _inherit = 'res.config.settings'
 
+    company_id = fields.Many2one('res.company', string='Company', required=True,
+        default=lambda self: self.env.user.company_id)
     sale_note = fields.Text(related='company_id.sale_note', string="Terms & Conditions")
     use_sale_note = fields.Boolean(
         string='Default Terms & Conditions',
@@ -69,7 +72,6 @@ class SaleConfigSettings(models.TransientModel):
         oldname='deposit_product_id_setting',
         help='Default product used for payment advances')
     auto_done_setting = fields.Boolean("Lock Confirmed Orders")
-    module_sale_subscription = fields.Boolean("Subscriptions")
     module_website_sale_digital = fields.Boolean("Sell digital products - provide downloadable content on your customer portal")
 
     auth_signup_uninvited = fields.Selection([
@@ -78,7 +80,6 @@ class SaleConfigSettings(models.TransientModel):
     ], string='Customer Account')
 
     group_multi_currency = fields.Boolean("Multi-Currencies", implied_group='base.group_multi_currency')
-    module_sale_stock = fields.Boolean("Inventory Management")
     module_delivery = fields.Boolean("Shipping Costs")
     module_delivery_dhl = fields.Boolean("DHL")
     module_delivery_fedex = fields.Boolean("FedEx")
@@ -86,10 +87,7 @@ class SaleConfigSettings(models.TransientModel):
     module_delivery_usps = fields.Boolean("USPS")
     module_delivery_bpost = fields.Boolean("bpost")
 
-    module_sale_timesheet = fields.Boolean("Timesheets")
-    module_sale_ebay = fields.Boolean("eBay")
     module_print_docsaway = fields.Boolean("Docsaway")
-    module_web_clearbit = fields.Boolean("Customer Autocomplete")
     module_product_email_template = fields.Boolean("Specific Email")
     module_sale_coupon = fields.Boolean("Coupons & Promotions")
 
@@ -144,7 +142,7 @@ class SaleConfigSettings(models.TransientModel):
             auth_signup_uninvited='b2c' if ICPSudo.get_param('auth_signup.allow_uninvited', 'False').lower() == 'true' else 'b2b',
             use_sale_note=ICPSudo.get_param('sale.use_sale_note', default=False),
             auto_done_setting=ICPSudo.get_param('sale.auto_done_setting'),
-            default_deposit_product_id=ICPSudo.get_param('sale.default_deposit_product_id'),
+            default_deposit_product_id=int(ICPSudo.get_param('sale.default_deposit_product_id')),
             sale_show_tax=ICPSudo.get_param('sale.sale_show_tax', default='subtotal'),
             multi_sales_price=sale_pricelist_setting in ['percentage', 'formula'],
             multi_sales_price_method=sale_pricelist_setting in ['percentage', 'formula'] and sale_pricelist_setting or False,
